src/lib/services/state-service.ts

```diff
@@ -712,26 +712,35 @@ export default class StateService {
         const v4 = await this.exportV4({ ...opts, includeEnvironments: true });
         // undefined defaults to true
         if (opts.includeFeatureToggles !== false) {
-            const keepEnv = v4.environments
-                .filter((env) => env.enabled !== false)
-                .sort((e1, e2) => {
-                    if (e1.type !== 'production' || e2.type !== 'production') {
-                        if (e1.type === 'production') {
-                            return -1;
-                        } else if (e2.type === 'production') {
-                            return 1;
-                        }
-                    }
-                    return e1.sortOrder - e2.sortOrder;
-                })[0];
-
-            const featureEnvs = v4.featureEnvironments.filter(
-                (fE) => fE.environment === keepEnv.name,
+            const enabledEnvironments = v4.environments.filter(
+                (env) => env.enabled !== false,
             );
+
+            const featureAndEnvs = v4.featureEnvironments.map((fE) => {
+                const envDetails = enabledEnvironments.find(
+                    (env) => fE.environment === env.name,
+                );
+                return { ...fE, ...envDetails, active: fE.enabled };
+            });
             v4.features = v4.features.map((f) => {
-                const variants = featureEnvs.find(
-                    (fe) => fe.featureName === f.name,
-                )?.variants;
+                const variants = featureAndEnvs
+                    .sort((e1, e2) => {
```

Comment:
The sorting of the environments is not absolute, it depends on whether the feature in a particular project has that environment active or not plus the other conditions.